tests/exceptions/clientException.test.ts

```diff
@@ -0,0 +1,63 @@
+import { AxiosError } from 'axios';
+import { ClientException } from '../../src';
+
+interface AxiosErrorTestData {
+  originalStatusCode: number;
+  exceptionStatusCode: number;
+}
+
+describe('ClientException', () => {
+  describe('converts Axios errors', () => {
+    const testServiceName = 'test-service-name';
+
+    const createAxiosError = (status: number): AxiosError => ({
+      isAxiosError: true,
+      config: {},
+      toJSON: () => ({}),
+      name: 'test axios error name',
+      message: 'test axios error message',
+      response: {
+        config: {},
+        headers: {},
+        request: {},
+        data: {},
+        statusText: 'test axios error status text',
+        status,
+      },
+    });
+
+    const testData: AxiosErrorTestData[] = [
+      {
+        originalStatusCode: 400,
+        exceptionStatusCode: 422,
+      },
+      {
+        originalStatusCode: 401,
+        exceptionStatusCode: 401,
+      },
+      {
+        originalStatusCode: 403,
+        exceptionStatusCode: 403,
+      },
+      {
+        originalStatusCode: 404,
+        exceptionStatusCode: 422,
+      },
+    ];
+
+    testData.map(({ exceptionStatusCode, originalStatusCode }) =>
+      test(`from ${originalStatusCode} to ${exceptionStatusCode}`, () => {
+        const axiosError = createAxiosError(originalStatusCode);
+        const clientException = new ClientException(testServiceName, axiosError);
+
+        expect(clientException.message).toEqual('Dependent service returned error');
+        expect(clientException.originalStatusCode).toEqual(originalStatusCode);
+        expect(clientException.statusCode).toEqual(exceptionStatusCode);
+        expect(clientException.details).toEqual({
+          serviceName: testServiceName,
+          error: {}, // this is empty due to how utils/serialize works
```

Comment:
@mmaruniak as the author of the `serialize` function ... do you expect this to be `{}` even though the used `AxiosError` had a lot of top-level properties?

Reply:
no

Reply:
We could probably add something if insteanceof Error then serialize as there it is adding properties

Reply:
Now I understand what's happening. It is because the `details` is an object of
```
serviceName: 'text',
error: { prop1: val1, prop2: val2 }
```
So the things inside `error` are getting cut out as they are not the top-level properties.

Is your proposal to check the top-level props and if they are instanceof Error then serialize them?

Reply:
Did some hackery, please re-check :)